Q.py

```diff
@@ -0,0 +1,123 @@
+import random
+import numpy as np
+import BFS
+import sokoban as sb
+import board as b
+import pprint
+
+# States
+DEAD = "Dead"
+ALIVE = "Alive"
+GOAL = "Goal"
+
+# Moves
+UP = "Up"
+DOWN = "Down"
+LEFT = "Left"
+RIGHT = "Right"
+
+# determine if any boxes are terminal states
+def is_terminal_state(board, boxes, rewards):
+
+    #if all boxes are in goals, terminal will remain true
+    for box in boxes:
+        # if the reward for this box's location is -100, then it is in a corner terminal state
+        if rewards[box[0], box[1]] == -100:
```

Comment:
is it possible we could get an out of bounds exception?  are we enforcing somewhere that the boxes parameter will always have tuple pairs of boxes?

Reply:
these are the box's x and y coordinates

Reply:
I still think this needs to be renamed.  one possible suggestion is 
for **coordinate** in boxes 

also add a comment above the check so we will be able to understand it when referencing later